File: location_based_game/backend/src/main/java/it/unicam/locationbasedgame/enums/Role.java
package it.unicam.locationbasedgame.enums;

/**
 * The kind of account.
 */
public enum Role {

    player("Player"),
    admin("Administrator");

    /** String label for the role. */
    private final String label;

    Role(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }
}

File: location_based_game/frontend/lib/enum/role.dart
/// The kind of account.
enum Role {
  player,
  admin;

  String get label => switch (this) {
        Role.player => 'Player',
        Role.admin => 'Administrator',
      };

  /// True when this account is allowed to open the admin menu.
  bool get canManageContent => this == Role.admin;
}